File: backend/schemas/story_request.py
from pydantic import BaseModel, Field
from typing import Optional

class StoryRequest(BaseModel):
    genre: str = Field(..., min_length=3)
    number_of_characters: Optional[int] = Field(default=None, gt=0)
    name_of_characters: Optional[list[str]] = Field(default=None)
    paragraphs: int = Field(default=1)
    generate_images: bool = Field(default=False)

File: frontend/lib/models/story_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'story_request.g.dart';

@JsonSerializable()
class StoryRequest {
  final String genre;
  final int? numberOfCharacters;
  final List<String>? nameOfCharacters;
  final int paragraphs;
  final bool generateImages;

  StoryRequest({
    required this.genre,
    this.numberOfCharacters,
    this.nameOfCharacters,
    this.paragraphs = 2, // Default to 2 paragraphs as per requirements
    this.generateImages = false,
  });

  factory StoryRequest.fromJson(Map<String, dynamic> json) =>
      _$StoryRequestFromJson(json);

  Map<String, dynamic> toJson() => _$StoryRequestToJson(this);
}
